### DISCUSSION TITLE
documentation is not good for reading ERA5 data

### DISCUSSION THREAD

**Discussion by aedessler** [microsoft/PlanetaryComputer#74] [General]
I'm a pretty experienced python programmer, but I cannot figure out how to read the ERA5 data.  You need to have better tutorials on this.  In particular, the PySTAC interface is completely opaque.  I normally am able to take a tutorial and modify it for my purposes, but there's not enough in the tutorial for me to do that.

**Comment by TomAugspurger** [comment]
Can you expand a bit on what's not good? And maybe provide an example of what you'd like to see?

Note that using STAC / pystac is completely optional. All of the data we host are in [Blob Storage](https://planetarycomputer.microsoft.com/docs/concepts/sas/) so you can access it directly from there if you'd like.

**Comment by aedessler** [comment]
The basic issue is that, while I'm an experienced python programmer, I'm not that knowledgeable about cloud computing.  Once I get the data into an xarray, I'm fine, but getting it there is the problem.  I usually work by taking example code and modifying it, but the example code (for both the ERA5 and cil-gdpcir-example) were very hard to figure out.  

I suppose what I'm looking for is more of a tutorial containing code snippets I can copy or better examples of how to read the various data sets.

I did eventually get it to work and it was not actually that hard.  The main issues I had were with the "search.catalog" command and then how to use what that returns to get data out of an xarray open_dataset command.  More info on both of those would be very very helpful.

**Reply by TomAugspurger** [reply]
> I suppose what I'm looking for is more of a tutorial containing code snippets I can copy or better examples of how to read the various data sets.

Each of our datasets has a notebook that, at a minimum, shows finding the data through the STAC API and loading it. Is that what you had in mind?

In addition to the per-dataset examples, we have a [STAC quickstart](https://planetarycomputer.microsoft.com/docs/quickstarts/reading-stac/) and pystac-client has its [own documentation](https://pystac-client.readthedocs.io/).

**Comment by aedessler** [comment]
First, let me say that the planetary computer is fantastic.  After a bit of work, I was able to figure out what I needed to do by following the example notebooks.  Ultimately, I was able to get my calculation done in a fraction of the time it would have taken me to have done it otherwise (you can see it here: https://interchange.puc.texas.gov/Documents/53401_8_1217222.PDF).  That said, perhaps my main suggestion would be to add examples.  

Other suggestions: I don't know how to use blob storage ... is there an example notebook showing how you can access the data that way?  Also, I asked about a tmp drive (on another question thread) and you said that I could store the data in blob storage.  Again, I'm not sure how to do that.  Do I have to pay to set that up?  Finally, I'm not sure what the easiest way to stage my output files so that I can download them.

Thanks again!

**Reply by TomAugspurger** [reply]
Best practices for using blob storage is indeed a weak spot in our documentation. In the meantime, https://pangeo.io/cloud.html#cloud-object-storage has some good background (anything that applies to S3 or Google Cloud Storage also applies to Azure Blob Storage in this context).

You would need your own Azure subscription to write to your own storage account. https://planetarycomputer.microsoft.com/docs/quickstarts/storage/ Has an example writing a COG to blob storage, but in general now you write it will depend on how you have your data loaded and what format you're writing.
